Empire/Empire.cpp

```diff
@@ -107,6 +112,38 @@ namespace {
         return retval;
     }
 
+    float CalculateNewStockpile(int empire_id, int stockpile_location_id, float starting_stockpile,
+                                         const std::map<std::set<int>, float>& available_pp,
+                                         const std::map<std::set<int>, float>& allocated_pp,
+                                         const std::map<std::set<int>, float>& allocated_stockpile_pp)
+    {
+        const Empire* empire = GetEmpire(empire_id);
+        if (!empire) {
+            ErrorLogger() << "CalculateStockpileContribution() passed null empire.  doing nothing.";
+            return 0.0f;
+        }
+        float stockpile_transfer_efficiency = empire->GetMeter("METER_IMPERIAL_PP_TRANSFER_EFFICIENCY")->Current();
+        // locally, within same resource group as the stockpile location (empire capital) we currently allow stockpile contribution at no penalty
+        float local_stockpile_transfer_rate = 1.0;
+        float remote_stockpile_transfer_rate = stockpile_transfer_efficiency;
+        
+        float stockpile_used = boost::accumulate(allocated_stockpile_pp | boost::adaptors::map_values, 0.0f);
+        float new_contributions = 0.0f;
+        for (auto const& available_group: available_pp) {
+            auto alloc_it = allocated_pp.find(available_group.first);
+            float allocated_here = (alloc_it == allocated_pp.end()) ? 0.0f : alloc_it->second;
+            float excess_here = available_group.second - allocated_here;
+            if (excess_here < EPSILON)
+                continue;
+            bool local_to_stockpile = available_group.first.find(stockpile_location_id) != available_group.first.end();
+            float this_transfer_rate = local_to_stockpile ? 
+                        local_stockpile_transfer_rate : remote_stockpile_transfer_rate;
```

Comment:
That was three tabs (12 spaces).  I could shorten it to two tabs / 8 spaces (which leaves it 1 space off from aligning with this_transfer_rate, or use 7 spaces to fully align it to this_transfer_rate, though frankly I think those both look more cluttered and less clearly a continuation line than the 12 spaces.  Please give me a specific instruction on how you want this indented.